src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -1,145 +1,99 @@
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = Configuration.getForceHttps();
-    this.forceLanguageInPath = Configuration.getForceLanguageInPath();
+  private UrlBuilder(String wiki, String env, Boolean forceHttps, Boolean forceLanguageInPath, String language) {
+    super(env);
+    this.wikiName = wiki;
+    this.forceHttps = forceHttps;
+    this.forceLanguageInPath = forceLanguageInPath;
+    this.language = language;
   }
 
-  public UrlBuilder(String env, Boolean forceHttps) {
-    this.env = env;
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = forceHttps;
-    this.forceLanguageInPath = Configuration.getForceLanguageInPath();
+  public static UrlBuilder createUrlBuilder() {
+    return createUrlBuilderForWikiAndLang(getWikiName(), getWikiLanguage());
   }
 
-  public UrlBuilder(String env, Boolean forceHttps, Boolean forceLanguageInPath) {
-    this.env = env;
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = forceHttps;
-    this.forceLanguageInPath = forceLanguageInPath;
+  public static UrlBuilder createUrlBuilderForWiki(String wiki) {
+    return createUrlBuilderForWikiAndLang(wiki, getWikiLanguage());
   }
 
-  public String getLanguageForWiki() {
-    return Configuration.getWikiLanguage() == null ? Configuration.DEFAULT_LANGUAGE : Configuration.getWikiLanguage();
+  public static UrlBuilder createUrlBuilderForWikiAndLang(String wiki, String language) {
+    return new UrlBuilder(wiki, getEnv(), getForceHttps(), getForceLanguageInPath(), language);
   }
 
   public String normalizePageName(String pageName) {
     return pageName.replace(" ", "_");
   }
 
   public String getUrlForPageWithWWW(String pageName) {
-    return getUrlForWiki(Configuration.getWikiName(), true) + pageName;
+    return getUrlForWiki(getWikiName(), true) + pageName;
   }
 
-  public String getUrlForPage(String pageName) {
+  public String getUrlForWikiPage(String pageName) {
     return getUrlForWiki() + pageName;
   }
 
-  public String getUrlForPage(String wikiName, String pageName) {
-    return getUrlForWiki(wikiName) + pageName;
-  }
-
-  public String getUrlForPage(Page page) {
-    if (page.getWikiPath() == null) {
-      return getUrlForWiki(page.getWikiName(), page.getWikiLanguage(), false);
-    }
-    return getUrlForPath(page.getWikiName(), page.getWikiLanguage(), page.getWikiPath());
-  }
-
-  public String getUrlForPage(Page page, String qs) {
-    return appendQueryStringToURL(page.getUrl(), qs);
-  }
-
-  public String getUrlForPath(String wikiName, String language, String wikiPath) {
-    return addPathToUrl(getUrlForWiki(wikiName, language, false, envType), wikiPath);
-  }
-
-  public String getUrlForPath(String wikiName, String wikiPath) {
-    return addPathToUrl(getUrlForWiki(wikiName), wikiPath);
-  }
+  //!!!!!!!!!!!!!!!!!!!Check that!!!!!!!!
+//  public String getUrlForPath(String language, String wikiPath) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)